swiftuploader/upload.py

```diff
@@ -108,29 +116,32 @@ def sizeof_fmt(num, suffix='B'):
         num /= 1024.0
     return "%.1f %s%s" % (num, 'Yi', suffix)
 
-
-def create_connection(auth_url, region, project_name, username, password):
-    prof = profile.Profile()
-    prof.set_region(profile.Profile.ALL, region)
-
-    conn = connection.Connection(
-        profile=prof,
-        user_agent='citrixswiftuploader',
-        auth_url=auth_url,
-        project_name=project_name,
-        username=username,
-        password=password
-    )
+def set_cloud_password(password, cloud_name):
+    main_path = os.path.expanduser('~')
+    conf_file = main_path + "/.config/openstack/clouds.yaml"
+    if not os.path.exists(conf_file):
+        conf_file = "/etc/openstack/clouds.yaml"
+    with open(conf_file) as f:
+        clouds_dict = yaml.load(f)
+        clouds_dict['clouds'][cloud_name]['auth']['password'] = password
+    with open(conf_file, "w") as f:
+        yaml.dump(clouds_dict, f)
+
+def create_connection(password, cloud_name):
+    set_cloud_password(password, cloud_name)
+    opts = cloud_conf(cloud_name)
+    occ = os_client_config.OpenStackConfig()
+    cloud = occ.get_one_cloud(opts.cloud)
+    conn = connection.from_config(cloud_config=cloud, options=opts)
+    set_cloud_password(PWD_COVER, cloud_name)
     conn.authorize()
```

Comment:
confused why this is needed.

Reply:
Just want to prevent the password to be seen, at most of the times...